sql/analyzer/aliases.go

```diff
@@ -303,3 +300,17 @@ func renameAliases(node sql.Node, oldNameLower string, newName string) (sql.Node
 		}
 	})
 }
+
+func disambiguateTableFunctions(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	var i int
+	return transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+		switch n := n.(type) {
+		case *expression.UnresolvedTableFunction:
+			i++
+			return plan.NewTableAlias(fmt.Sprintf("%s_%d", n.Name(), i), n), transform.NewTree, nil
```

Comment:
To avoid an unlikely name collision, we should get all the aliases for the node, then check if the new name conflicts with any existing names.

Reply:
Giving every relational node its own unique id is one of my first TODOs after getting nautobot running.